aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -0,0 +1,198 @@
+import software.amazon.awssdk.services.glue.model.Table;
+import software.amazon.awssdk.services.glue.model.TableInput;
+import software.amazon.awssdk.services.glue.model.UpdateTableRequest;
+
+class GlueTableOperations extends BaseMetastoreTableOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GlueTableOperations.class);
+
+  // same as org.apache.hadoop.hive.metastore.TableType.EXTERNAL_TABLE
+  // more details: https://docs.aws.amazon.com/glue/latest/webapi/API_TableInput.html
+  private static final String GLUE_EXTERNAL_TABLE_TYPE = "EXTERNAL_TABLE";
+
+  private final GlueClient glue;
+  private final AwsProperties awsProperties;
+  private final String databaseName;
+  private final String tableName;
+  private final String fullTableName;
+  private final FileIO fileIO;
+
+  GlueTableOperations(GlueClient glue, String catalogName, AwsProperties awsProperties,
+                      FileIO fileIO, TableIdentifier tableIdentifier) {
+    this.glue = glue;
+    this.awsProperties = awsProperties;
+    this.databaseName = IcebergToGlueConverter.getDatabaseName(tableIdentifier);
+    this.tableName = IcebergToGlueConverter.getTableName(tableIdentifier);
+    this.fullTableName = String.format("%s.%s.%s", catalogName, databaseName, tableName);
+    this.fileIO = fileIO;
+  }
+
+  @Override
+  public FileIO io() {
+    return fileIO;
+  }
+
+  @Override
+  protected String tableName() {
+    return fullTableName;
+  }
+
+  @Override
+  protected void doRefresh() {
+    String metadataLocation = null;
+    Optional<Table> tableOptional = getGlueTable();
+    if (tableOptional.isPresent()) {
+      Table table = tableOptional.get();
+      GlueToIcebergConverter.validateTable(table, tableName());
+      metadataLocation = table.parameters().get(METADATA_LOCATION_PROP);
+    } else {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchTableException("Cannot find Glue table %s after refresh, " +
+            "maybe another process deleted it or revoked your access permission", tableName());
+      }
+    }
+
+    refreshFromMetadataLocation(metadataLocation);
+  }
+
+  @Override
+  protected void doCommit(TableMetadata base, TableMetadata metadata) {
+    String newMetadataLocation = writeNewMetadata(metadata, currentVersion() + 1);
+    boolean exceptionThrown = true;
+    boolean isUpdate = false;
+    Table glueTable = null;
+    try {
+      Optional<Table> glueTableOptional = getGlueTable();
+      if (glueTableOptional.isPresent()) {
+        glueTable = glueTableOptional.get();
+        isUpdate = true;
+        // If we try to create the table but the metadata location is already set, then we had a concurrent commit
+        if (base == null && glueTable.parameters().get(METADATA_LOCATION_PROP) != null) {
+          throw new AlreadyExistsException("Cannot commit because table %s already exists in Glue", tableName());
+        }
+      }
+
+      checkMetadataLocation(isUpdate, glueTable, base);
```

Comment:
If the above block is removed, then this no longer needs to be in the `try` block. I think that also means you can remove `CommitFailedException` from the `catch`.